app/controllers/api/results_controller.rb

```diff
@@ -6,6 +6,44 @@ def results_search_conditions
       MiqReportResult.for_user(User.current_user).where_clause.ast
     end
 
+    def sort_order
+      params['sort_order'] == 'desc' ? :descending : :ascending
+    end
+
+    def param_result_set?
+      params.key?(:hash_attribute) && params[:hash_attribute] == "result_set"
+    end
+
+    def report_options
+      params.merge(:sort_by => params['sort_by'], :sort_order => sort_order).merge(filter_options)
+    end
+
+    def filter_options
+      filtering_enabled? ? {:filter_string => params[:filter_string], :filter_column => params[:filter_column]} : {}
+    end
+
+    def filtering_enabled?
+      params.key?(:filter_column) && params.key?(:filter_string) && params[:filter_string]
+    end
```

Comment:
can you move sort_order, param_result_set?, report_options, filter_options and filtering_enabled? to app/controllers/api/base_controller/parameter.rb ?

Also, can't we leverage the sort_params method from that file ?

Reply:
> can you move sort_order, param_result_set?, report_options, filter_options and filtering_enabled? to app/controllers/api/base_controller/parameter.rb ?

Yes, done 👍 and I put it to special module.

> 
> Also, can't we leverage the sort_params method from that file ?

it is not possible (if we will not change sort_params somehow), especially because we don't to return from [AREL](https://github.com/manageiq/manageiq-api/blob/713b792ca06baf0be0c610b91c47940539717a81/app/controllers/api/base_controller/parameters.rb) for direction or sorting - sorting in this PR is on array. 

 @abellotti  otherwise thanks for review,
 is it ok now ?

Reply:
Ahh, that makes sense.  Thanks! !